### PR TITLE
Move logic for facebook count updates to the model and improve it

### PR DESCRIPTION
**Pull request by gaganawhad**
Improvements include guards against decreasing the count or
niling / zeroing them.

### CODE REVIEW

**Comment by gaganawhad** [comment]
[Update Facebook Share Count Service to Handle API Failures Better](https://trello.com/c/bLY1f6hR/2925-update-facebook-share-count-service-to-handle-api-failures-better)

File: spec/models/wakes/location_spec.rb [hunk 1]
```diff
@@ -61,4 +61,69 @@
...
+
+    context 'when count is greater than the current count' do
+      before { location.update_facebook_count(101) }
+
+      it 'updates facebook_count' do
+        expect(location.facebook_count).to eq 101
+      end
+
+      it 'updates facebook_count_updated_at' do
+        expect(location.facebook_count_updated_at).to eq Time.zone.now
```
Comment (benhutton) [inline spec/models/wakes/location_spec.rb]: @gaganawhad can you add parens around `Time.zone.now`? Too many spaces

File: app/models/concerns/wakes/metrics/facebook.rb [hunk 2]
```diff
@@ -15,6 +15,15 @@ def enqueue_facebook_count_update
         def self.ordered_for_facebook_updates
           order("document->'facebook_count_updated_at' ASC NULLS FIRST")
         end
+
+        def update_facebook_count(count)
+          if facebook_count.to_i <= count.to_i
```
Comment (benhutton) [inline app/models/concerns/wakes/metrics/facebook.rb]: @gaganawhad 2 things:

1. can you rename `count` to something like `new_facebook_count`? Given you compare it with the old count, having the names be very similar will communicate more clearly what's going on.

2. can you flip this around? You are testing the new count, not the old count. Think of `<=` as a method call. Seems more clear to me to call that method on the new candidate count, not the old one.

So, all together: `if new_facebook_count.to_i >= facebook_count.to_i`

**Comment by gaganawhad** [comment]
@benhutton take another look.

File: spec/models/wakes/location_spec.rb (hunk 1 above)
Comment (benhutton) [inline spec/models/wakes/location_spec.rb]: @gaganawhad you got some of these, but not all

**Comment by gaganawhad** [comment]
@benhutton how about now?

**Pull request merged**